compile error with avr-rust

```
error[E0119]: conflicting implementations of trait `array::Array` for type `[_; 0]`:
  --> /Users/wez/.cargo/registry/src/github.com-1ecc6299db9ec823/arrayvec-0.4.5/src/array.rs:65:9
   |
65 |            unsafe impl<T> Array for [T; $len] {
   |   _________^
   |  |_________|
   | ||
66 | ||             type Item = T;
67 | ||             type Index = $index_type;
68 | ||             #[inline(always)]
...  ||
73 | ||             fn capacity() -> usize { $len }
74 | ||         }
   | ||         ^
   | ||_________|
   | |__________first implementation here
   |            conflicting implementation for `[_; 0]`
...
90 |    fix_array_impl_recursive!(u32, 1 << 16,);
   |    ----------------------------------------- in this macro invocation

error: aborting due to previous error

error: Could not compile `arrayvec`.
```

commenting out the 32-bit index case in `array.rs:90` resolves this compilation error.

I'm not sure of the best way to resolve this otherwise I'd just submit a PR to do it :-/

--- a/src/array.rs
+++ b/src/array.rs
@@ -87,5 +87,7 @@ fix_array_impl_recursive!(u8, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 
                           16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31,
                           32, 40, 48, 50, 56, 64, 72, 96, 100, 128, 160, 192, 200, 224,);
 fix_array_impl_recursive!(u16, 256, 384, 512, 768, 1024, 2048, 4096, 8192, 16384, 32768,);
+// This array size doesn't exist on 16-bit
+#[cfg(any(target_pointer_width="32", target_pointer_width="64"))]
 fix_array_impl_recursive!(u32, 1 << 16,);
 

